core/jazz_t-vault/index.js

```diff
@@ -0,0 +1,499 @@
+}
+
+function updateSafe(safeDetails, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    safeDetails.safeName = safeName;
+    utils.updateSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function createUserInSafe(safeDetails, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    safeDetails.safeName = safeName;
+    utils.createUserInSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function deleteUserFromSafe(safeDetails, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    safeDetails.safeName = safeName;
+    utils.deleteUserFromSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function getRoleInSafe(roleName, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    let safeDetails = {};
+    safeDetails.safeName = safeName;
+    safeDetails.roleName = roleName;
+    utils.getRoleInSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function createRoleInSafe(safeDetails, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    safeDetails.safeName = safeName;
+    utils.createRoleInSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function deleteRoleFromSafe(safeDetails, safeName, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    safeDetails.safeName = safeName;
+    utils.deleteRoleFromSafe(safeDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function createUserInVault(userDetails, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    utils.createUserInVault(userDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+function deleteUserFromVault(userDetails, configData, vaultToken) {
+  return new Promise((resolve, reject) => {
+    utils.deleteUserFromVault(userDetails, configData, vaultToken, function (err, data) {
+      if (err) {
+        return reject(err);
+      } else {
+        return resolve(data);
+      }
+    });
+  });
+}
+
+const exportable = {
+  handler,
+  validateCreateSafeInput,
```

Comment:
Move all the validation related functions to `utils/validations.js`?

Reply:
Done